enhancements/ingress/route-secret-injection-for-external-certificate-management.md

```diff
@@ -0,0 +1,375 @@
+
+**When an Ingress is used to expose user workload**
+- The end user must have generated the serving certificate as a pre requisite
+  using third-party systems like cert-manager.
+- In cert-manager's case, the Certificate CR must be created in the same namespace
+  where the Ingress is going to be created.
+- To expose a user workload, he would need to create a new Ingress with the generated secret
+  referenced in `.spec.tls.secretName`.
+- If the secret CR that is referenced exists and has a successfully generated
+  cert/key pair generated, the ingress-to-route controller copies this certificate data 
+  into the route at `.spec.tls.certificate` and `.spec.tls.key`.
+
+**When a Route is directly used to expose user workload**
+- The end user must have generated the serving certificate as a pre requisite
+  using third-party systems like cert-manager.
+- To expose a user workload, the user would create a new Route with an annotation
+  referencing the generated secret that was created in the previous step.
+- If the secret that is referenced exists and has a successfully generated
+  cert/key pair generated, this certificate data is copied into the route at 
+  `.spec.tls.certificate` and `.spec.tls.key`.
+
+*Note*: The same workflow would apply to updating an existing Route to integrate
+with cert-manager. The new certificate data will overwrite the `.spec.tls.certificate` 
+and `.spec.tls.key` in the route CR since the annotation gets a higher preference.
+
+#### Certificate Life cycle
+
+Post integrating with third-party solutions like cert-manager, the end user can also
+disable automatic certificate management. This can be done in 2 ways,
+- The end user can delete the annotation on the Route CR.
+- The end user deletes the secret that was associated with the Certificate CR.
+
+**Deleting the annotation on the Route CR**
+The end user can delete the annotation on the Route CR, this will result
+in abondoning the cert/key pair that was generated via cert-manager and resort to
+using the default certificates that were initially generated by the cluster-ingress-operator
+i.e. clearing the `.spec.tls` on Route. 
+
+**Deleting the generated secret that contains the serving cert/key pair**
+When the end user deletes the secret containing the certificate data, the `.spec.tls` on 
+Route will be cleared. This will result in the route serving default certificates. In such
+a scenario the annotation that was added by the user will be automatically deleted by the 
+secret-injector controller. This is only in the case where the user is directly creating 
+routes instead of relying on the Ingresses.
+
+#### Variation [optional]
+
+N.A
+
+### API Extensions
+
+A new annotation of type `string` to be introduced as part of the integration
+to support third-party certificate management systems,
+
+```yaml
+annotations:
+  route.openshift.io/tls-secret-reference: <secret-name>
+```
+
+*Note*: The default value would be N/A.
```

Comment:
The expected contents of the secret as well as  the assumed namespace should be described too.

Reply:
Addressed